### DISCUSSION TITLE
Actions-ecosystem/action-remove-labels fails "Resource not accessible by integration"

### DISCUSSION THREAD

**Discussion by gianarb** [community/community#25937] [Actions]
<p>Hello</p>
<p>I am able to trigger a job only when a particular label gets added to the pull request.</p>
<p>As first step I would like to remove the label.</p>
<p>This is the workflows file:</p>
<pre><code class="lang-yaml">name: Setup with Vagrant on Packet
on:
  push:
  pull_request:
    types: [labeled]

jobs:
  vagrant-setup:
    if: contains(github.event.pull_request.labels.*.name, 'ci-check/vagrant-setup')
    runs-on: self-hosted
    steps:
    - uses: actions-ecosystem/action-remove-labels@v1
      with:
        github_token: ${{ secrets.github_token }}
        labels: ci-check/vagrant-setup
    - name: Checkout
      uses: actions/checkout@v2
    - name: Vagrant Test
      run: |
        export VAGRANT_DEFAULT_PROVIDER="virtualbox"
        go test -v ./test/_vagrant
</code></pre>
<p>And this is the output I get from the <code>actions-ecosystem/action-remove-labels@v1</code> step:</p>
<pre><code class="lang-auto">2020-07-23T13:51:36.8848935Z ##[group]Run actions-ecosystem/action-remove-labels@v1
2020-07-23T13:51:36.8849248Z with:
2020-07-23T13:51:36.8849902Z   github_token: ***
2020-07-23T13:51:36.8850086Z   labels: ci-check/vagrant-setup
2020-07-23T13:51:36.8850285Z   repo: tinkerbell/tink
2020-07-23T13:51:36.8850456Z ##[endgroup]
2020-07-23T13:51:38.0240570Z ##[error]HttpError: Resource not accessible by integration
2020-07-23T13:51:38.0243953Z ##[error]Resource not accessible by integration
</code></pre>
<p>Thanks</p>

**Comment by gianarb** [comment]
<p>The PR comes from a fork and I think the GITHUB_TOKEN only has read permission if workflows comes from a fork.</p>
<p>So I created a new personal access token and I place it as a secret, and I have updated my step:</p>
<pre><code class="lang-yaml">    - uses: actions-ecosystem/action-remove-labels@v1
      with:
        github_token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
        labels: ci-check/vagrant-setup
</code></pre>
<p>Now the error I get is:</p>
<pre><code class="lang-auto"> Input required and not supplied: github_token
</code></pre>
<p>Probably because the secret is empty probably because of this doc</p>
<pre><code class="lang-auto">With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository.
</code></pre>
<p>Not sure how to get around it :tired_face: This is an open source project, people will mainly contribute via FORK and that’s why we are using a label as a gatekeeper to trigger a particular action</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/gianarb">@gianarb</a>,</p>
<p>As the docs has mentioned, when a workflow is triggered from a forked repository:</p>
<ol>
<li>
<p><a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow" rel="nofollow noopener">With the exception of <strong>GITHUB_TOKEN</strong>, secrets are not passed to the runner.</a></p>
</li>
<li>
<p>The <strong>GITHUB_TOKEN</strong> only has ‘<strong>read</strong>’ permission. For more information, see “<a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#permissions-for-the-github_token" rel="nofollow noopener"><strong>Permissions for the GITHUB_TOKEN</strong></a>”.</p>
</li>
</ol>
<p>If you want the action ‘<strong>actions-ecosystem/action-remove-labels@v1</strong>’ you are using in your workflow can remove labels from the pull request, the token used to authenticate must have the ‘<strong>write</strong>’ permission for pull request in the repository.<br>
However, as mentioned above, due to the workflow is triggered from a forked repository, there is not any available and safe way to pass the token which has the ‘<strong>write</strong>’ permission to the runner.</p>
<p>There is a good news is that the appropriate engineering team is planing to add an option “<strong>Send secrets to pull request workflows from forks</strong>” to the settings for <strong>Actions permissions</strong>.<br>
This option will allow the owners and administrators to decide whether to pass secrets to the runner when the workflow is triggered from a forked repository.<br>
This feature may be released recently. You can follow the <a href="https://github.blog/changelog/" rel="nofollow noopener"><strong>GitHub Changelog</strong></a> to view the release of latest features.</p>
